articles/react-pwa/index.md

```diff
@@ -0,0 +1,508 @@
+We are using the `useState` hook to return two values in an array. The first value is the state, and the second value is the function to update the state. We have named the destructured values as `inputText` and `setInputText`.
+Inside the form, we are binding the input tag to an event. When the state of the form changes, The method `inputTextHandler` is invoked and updates the state to the current input value by calling the `setInputText` function.
+The form is a controlled component that will track the `inputText`. The `submitTodoHandler` is bound on the form when a button is clicked that will invoke the `addTodo` function. We passed the `addTodo` down as props from the `App` component. The `setInputText` inside the `submitTodoHandler` will simply reset the form.
+We now need to add this component in our parent `App` component and pass the function `addTodo`.
+Below is the `App.js` file.
+```javascript
+import React,{useState} from 'react';
+
+import Form from './components/Form';
+
+const App=()=>{
+
+  const addTodo= todo=>{
+          const newTodos=[todo,...todos];
+           setTodos(newTodos);
+       }
+  
+  return (
+    <div>
+      <h1>TODO PWA</h1>
+      <Form addTodo={addTodo}/>
+    </div>
+  );
+}
+
+export default App;
+```
+
+### Creating the TodoList and Todo Components
+You need to create a new file named `TodosList.js` inside the components folder.
+This is the component that acts as a container to the list of the todo items.
+The state is in the App component and passed to the TodoList as props.
+Before creating the TodoList component, the App component has its state, so below is the entire code for the `App.js` file:
+A Todo component will represent a single todo item for each row
+with buttons to mark it as completed and delete it. The TodoList will be the container for the list of todos available.
+Since our Todo component is a child of the TodoList component, We will first create the TodoList component.
+First, we need to add the todos state inside the `App.js` file. So we will add extra props to the Form component.
+The todos state is [lifted](https://reactjs.org/docs/lifting-state-up.html) in the App component so that our `TodoList` will receive it via the props pattern.
+The `App.js` file is now as follows:
+
+```javascript
+import React,{useState} from 'react';
+
+import Form from './components/Form';
+import TodoList from './components/TodoList';
+
+const App=()=>{
+
+  const [todos,setTodos]=useState([]);
+
+  const addTodo= todo=>{
+          const newTodos=[todo,...todos];
+           setTodos(newTodos);
+       }
+  
+  return (
+    <div>
+      <h1>TODO PWA</h1>
+      <Form addTodo={addTodo}/>
+      <TodoList todos={todos} setTodos={setTodos}/>
+    </div>
+  );
+}
+
+export default App;
+```
+
+The todo will be an object with a structure like:
+```javascript
+    {
+      id:v4(),
+      task,
+      complete:false
+    }
+```    
```

Comment:
You should explain this in the Form component since you've already introduced it there.